client/client.go

```diff
@@ -50,41 +52,51 @@ func (c *Client) SendRequest(method string, path string, payload interface{}, st
 
 	req, err := http.NewRequest(method, url, b)
 	if err != nil {
-		return "", err
+		return "", "", err
 	}
 
 	req.SetBasicAuth(c.username, c.password)
 	req.Header.Add("Content-Type", "application/json")
 
 	resp, err := client.Do(req)
 	if err != nil {
-		return "", err
+		return "", "", err
 	}
 
 	body, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
-		return "", err
+		return "", "", err
 	}
 	resp.Body.Close()
 
 	strbody := string(body)
 
-	location := resp.Header.Get("location")
-	if location != "" {
-		id := strings.Replace(location, "/api/v2.0", "", -1)
-
-		localation := map[string]string{"localation": id}
-		json, _ := json.Marshal(localation)
-		strbody = string(json)
-
+	respHeaders := resp.Header
+	headers, err := json.Marshal(respHeaders)
+	if err != nil {
+		return "", "", err
 	}
 
 	if statusCode != 0 {
 		if resp.StatusCode != statusCode {
 
-			return "", fmt.Errorf("[ERROR] unexpected status code got: %v expected: %v \n %v", resp.StatusCode, statusCode, strbody)
+			return "", "", fmt.Errorf("[ERROR] unexpected status code got: %v expected: %v \n %v", resp.StatusCode, statusCode, strbody)
 		}
 	}
 
-	return strbody, nil
+	return strbody, string(headers), nil
+}
+
+// GetID gets the resource id from locaation resposne header
+func GetID(body string) (id string) {
+	var jsonData models.ResponseHeaders
+	json.Unmarshal([]byte(body), &jsonData)
```

Comment:
Remember to check for errors

Reply:
lol I knew you were going to ask about this. But it was getting me the below error
`cannot use err (variable of type error) as string value in return statement`

Reply:
Dont worry I worked it out.
`func GetID(body string) (id string, err error) {
	var jsonData models.ResponseHeaders
	err = json.Unmarshal([]byte(body), &jsonData)
	if err != nil {
		return "", err
	}

	localation := jsonData.Location[0]
	localation = strings.Replace(localation, "/api", "", -1)

	// removes /v2.0 from string if using api version 2
	localation = strings.Replace(localation, "/v2.0", "", -1)

	return localation, nil
}
`